fastapi_websocket_rpc/websocket_rpc_client.py

```diff
@@ -109,29 +107,34 @@ def __init__(self, uri: str, methods: RpcMethodsBase = None,
 
     async def __connect__(self):
         try:
-            # Make sure we don't have any hanging tasks (from previous retry runs)
-            self.cancel_tasks()
-            logger.info(f"Trying server - {self.uri}")
-            # Start connection
-            self.conn = websockets.connect(self.uri, **self.connect_kwargs)
-            # Get socket and wrap in our serialization class
-            raw_ws = await self.conn.__aenter__()
-            self.ws = self._serializing_socket_cls(raw_ws)
-            # Wrap
-            # Init an RPC channel to work on-top of the connection
-            self.channel = RpcChannel(self.methods, self.ws, default_response_timeout=self.default_response_timeout)
-            # register handlers
-            self.channel.register_connect_handler(self._on_connect)
-            self.channel.register_disconnect_handler(self._on_disconnect)
-            # Start reading incoming RPC calls
-            self._read_task = asyncio.create_task(self.reader())
-            # start keep alive (if enabled i.e. value isn't 0)
-            self._start_keep_alive_task()
-            # Wait for RPC channel on the server to be ready (ping check)
-            await self.wait_on_rpc_ready()
-            # trigger connect handlers
-            await self.channel.on_connect()
-            return self
+            try:
+                logger.info(f"Trying server - {self.uri}")
+                # Start connection
+                raw_ws = await websockets.connect(self.uri, **self.connect_kwargs)
+                # Wrap socket in our serialization class
+                self.ws = self._serializing_socket_cls(raw_ws)
```

Comment:
is `await self.conn.__aenter__()` equivalent to `await websockets.connect(self.uri, **self.connect_kwargs)`

Reply:
Yes - according to docs, `websockets.connect` can be either awaited or used as a context manager.